### ISSUE TITLE
make-socks-proxied-conn-manager broken in 3.3.0

### ISSUE THREAD

**Issue by ghuo** [dakrone/clj-http#334]
Trying this example:

```
  (in-ns 'clj-http.client)                                                                    
  (get "https://google.com"                                                                   
       {:connection-manager                                                                   
        (conn/make-socks-proxied-conn-manager "localhost" 17806)}) 
```

Yields:

```
java.lang.IllegalArgumentException: No matching method found: connectSocket for class clj_http.conn_mgr.proxy$org.apache.http.conn.ssl.SSLConnectionSocketFactory$ff19274a
 at clojure.lang.Reflector.invokeMatchingMethod (Reflector.java:53)
    clojure.lang.Reflector.invokeInstanceMethod (Reflector.java:28)
    clj_http.conn_mgr$SSLGenericSocketFactory$fn__15847$fn__15848.invoke (conn_mgr.clj:36)
    clojure.core$proxy_call_with_super.invokeStatic (core_proxy.clj:368)
    clojure.core$proxy_call_with_super.invoke (core_proxy.clj:364)
    clj_http.conn_mgr$SSLGenericSocketFactory$fn__15847.invoke (conn_mgr.clj:36)
    clj_http.conn_mgr.proxy$org.apache.http.conn.ssl.SSLConnectionSocketFactory$ff19274a.connectSocket (:-1)
```

I'll take a look at updating conn-mgr, but I'm not sure I have enough time to dig in now. Any context would be appreciated.

**Comment by dakrone** [comment]
I think this has to do with the upgrade to the non-deprecated Apache
APIs. There aren't any tests for the socks proxied connection manager,
which is why this wasn't caught until now.


--
;; Lee

**Comment by dakrone** [comment]
Should be fixed by #353
